src/Compilers/VisualBasic/Test/Emit/Attributes/InternalsVisibleToAndStrongNameTests.vb

```diff
@@ -913,18 +925,83 @@ BC31535: Friend assembly reference 'WantsIVTAccess' is invalid. Strong-name sign
 #End Region
 
 #Region "Signing"
+
+    <Fact>
+    Public Sub MaxSizeKey()
+        Dim pubKey = TestResources.General.snMaxSizePublicKeyString
+        Const pubKeyToken = "1540923db30520b2"
+        Dim pubKeyTokenBytes = AssemblyIdentity.CalculatePublicKeyToken(
+                ImmutableArray.Create(TestResources.General.snMaxSizePublicKey))
+
+        Dim comp = CreateCompilationWithMscorlib(
+<compilation>
+    <file name="c.vb">
+Imports System
+Imports System.Runtime.CompilerServices
+
+&lt;Assembly:InternalsVisibleTo("MaxSizeComp2, PublicKey=<%= pubKey %>, PublicKeyToken=<%= pubKeyToken %>")&gt;
+
+Friend Class C
+    Public Shared Sub M()
+        Console.WriteLine("Called M")
+    End Sub
+End Class
+    </file>
+</compilation>,
+                options:=TestOptions.ReleaseDll.WithCryptoKeyFile(SigningTestHelpers.MaxSizeKeyFile).WithStrongNameProvider(s_defaultProvider))
+
+        comp.VerifyEmitDiagnostics()
+
+        Assert.True(comp.IsRealSigned)
+        VerifySigned(comp)
+        Assert.Equal(TestResources.General.snMaxSizePublicKey, comp.Assembly.Identity.PublicKey)
+        Assert.Equal(Of Byte)(pubKeyTokenBytes, comp.Assembly.Identity.PublicKeyToken)
+
+        Dim comp2 = CreateCompilationWithMscorlib(
+<compilation name="MaxSizeComp2">
+    <file name="c.vb">
+Class D
+    Public Shared Sub Main()
+        C.M()
+    End Sub
+End Class
+    </file>
+</compilation>, references:={comp.ToMetadataReference()},
+options:=TestOptions.ReleaseExe.WithCryptoKeyFile(SigningTestHelpers.MaxSizeKeyFile).WithStrongNameProvider(s_defaultProvider))
+
+        CompileAndVerify(comp2, expectedOutput:="Called M")
+        Assert.Equal(TestResources.General.snMaxSizePublicKey, comp2.Assembly.Identity.PublicKey)
+        Assert.Equal(Of Byte)(pubKeyTokenBytes, comp2.Assembly.Identity.PublicKeyToken)
+
+        Dim comp3 = CreateCompilationWithMscorlib(
+<compilation name="MaxSizeComp2">
+    <file name="c.vb">
+Class D
+    Public Shared Sub Main()
+        C.M()
+    End Sub
+End Class
+    </file>
+</compilation>, references:={comp.EmitToImageReference()},
+options:=TestOptions.ReleaseExe.WithCryptoKeyFile(SigningTestHelpers.MaxSizeKeyFile).WithStrongNameProvider(s_defaultProvider))
+
+        CompileAndVerify(comp3, expectedOutput:="Called M")
+        Assert.Equal(TestResources.General.snMaxSizePublicKey, comp3.Assembly.Identity.PublicKey)
+        Assert.Equal(Of Byte)(pubKeyTokenBytes, comp3.Assembly.Identity.PublicKeyToken)
+    End Sub
+
     <Fact>
     Public Sub SignIt()
         Dim other As VisualBasicCompilation = CreateCompilationWithMscorlib(
 <compilation name="Sam">
-    <file name="a.vb"><![CDATA[
+                                    <file name="a.vb"><![CDATA[
```

Comment:
This file has a lot of whitespace changes that do not look appropriate. Please revert them.